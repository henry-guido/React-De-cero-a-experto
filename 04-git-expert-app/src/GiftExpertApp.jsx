import { useState } from 'react';
import {AddCategory, GifGrid} from './components'

export const GiftExpertApp = () => {
  const [categories, setCategories] = useState(['one punch']);

  const onAddCategory = (val) => {
    setCategories([val, ...categories]);
  };

  return (
    <>
      <h1>GifExpertApp</h1>
      <AddCategory onAddCategory={onAddCategory} />
      {categories.map((category) => (
        <div key={category} className='card-grid'>
          <GifGrid  category={category} />
        </div>
      ))}
    </>
  );
};
