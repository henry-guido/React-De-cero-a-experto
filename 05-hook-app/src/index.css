body{
  padding: 20px;
  background-color: #444;
  filter: invert();
}

button{
  margin-right: 10px;
}

nav{
  display: flex;
  justify-content: space-between;
}

ul{
  display: flex;
  align-items: center;
  margin: 0;
  gap: 10px
}

li{
  list-style: none;
}

a{
  text-decoration: none;
}