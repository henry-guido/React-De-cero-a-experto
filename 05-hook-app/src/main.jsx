import ReactDOM from 'react-dom/client';
import { BrowserRouter} from 'react-router-dom'
// import HookApp from './HookApp'
// import { CounterApp } from './01-useState/CounterApp'

import './index.css';
import { Padre } from './07-tarea-memo/Padre';
// import { SimpleForm } from './02-useEffect/SimpleForm'
// import { FormWithCustomHook } from './02-useEffect/FormWithCustomHook';
// import { MultipleCustomHooks } from './03-examples/MultipleCustomHooks';
// import { FocusScreen } from './04-useRef/FocusScreen';
// import { CounterWithCustomHook } from './01-useState/CounterWithCustomHook'
// import {Layout} from './05-useLayoutEffect/Layout'
// import './08-useReducer/intro-reducer'
// import { TodoApp } from './08-useReducer/TodoApp';
import { MainApp } from './09-useContext/MainApp';

ReactDOM.createRoot(document.getElementById('root')).render(
  <BrowserRouter>
   {/* <React.StrictMode> */}
  <MainApp />
  {/* </React.StrictMode>, */}
  </BrowserRouter>
);
