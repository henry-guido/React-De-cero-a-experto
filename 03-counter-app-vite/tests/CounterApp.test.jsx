import { fireEvent, render, screen } from "@testing-library/react"
import { CounterApp } from "../src/CounterApp"

describe('Prueba en <CounterApp />', () => { 
    const value = 100
    test('Debe hacer match con snapshot', () => { 
        const {container}  = render(<CounterApp value={value} />)
        expect(container).toMatchSnapshot()
     })

     test('Debe de mostrar el valor inicial de 100', () => { 
        render(<CounterApp value={value}/>)
        expect(screen.getByText(value)).toBeTruthy()
      })

      test('Debe incrementar con el botón +1', () => { 
        render(<CounterApp value={value} />)
        fireEvent.click( screen.getByText('+1') )
        expect( screen.getByText('101')).toBeTruthy()

       })
       
      test('Debe incrementar con el botón -1', () => { 
        render(<CounterApp value={value} />)
        fireEvent.click( screen.getByText('-1') )
        expect( screen.getByText('99')).toBeTruthy()

       })

      test('Debe de funcionar el boton de reset', () => { 
        render(<CounterApp value={value} />)
        fireEvent.click( screen.getByText('+1'))
        fireEvent.click( screen.getByText('+1'))
        fireEvent.click( screen.getByText('+1'))

        fireEvent.click(screen.getByRole('button', {name: 'btn-reset'}))
        expect( screen.getByText(100)).toBeTruthy()

       })
 })