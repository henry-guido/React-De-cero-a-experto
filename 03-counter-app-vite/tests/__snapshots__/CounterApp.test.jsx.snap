// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <CounterApp /> Debe hacer match con snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
     
    100
     
  </h2>
  <button>
     +1 
  </button>
  <button>
     -1 
  </button>
  <button
    aria-label="btn-reset"
  >
     Reset 
  </button>
</div>
`;
